Sources/_MatchingEngine/Regex/Parse/LexicalAnalysis.swift

```diff
@@ -550,6 +550,11 @@ extension Source {
         try src.expect("}")
         return opt
 
+      // Swift semantic level options
+      case "X": return advanceAndReturn(.graphemeClusterSemantics)
+      case "u": return advanceAndReturn(.unicodeScalarSemantics)
+      case "b": return advanceAndReturn(.byteSemantics)
+        
```

Comment:
These would be Swift-specific flags. The `X` flag is to match the `\X` metacharacter, which matches a grapheme cluster; given that, it might be better to go with `O` for Unicode scalar semantics to match `\O`. I'll go through this when I write up the flags.